sdk/src/userMap/userMap.ts

```diff
@@ -435,6 +449,135 @@ export class UserMap implements UserMapInterface {
 		}
 	}
 
+	private async paginatedSync() {
+		if (this.syncPromise) {
+			return this.syncPromise;
+		}
+
+		this.syncPromise = new Promise<void>((resolve) => {
+			this.syncPromiseResolver = resolve;
+		});
+
+		try {
+			const accountsPrefetch = await this.connection.getProgramAccounts(
+				this.driftClient.program.programId,
+				{
+					dataSlice: { offset: 0, length: 0 },
+					filters: [
+						getUserFilter(),
+						...(!this.includeIdle ? [getNonIdleUserFilter()] : []),
+					],
+				}
+			);
+			const accountPublicKeys = accountsPrefetch.map(
+				(account) => account.pubkey
+			);
+
+			const limitConcurrency = async (tasks, limit) => {
+				const executing = [];
+				const results = [];
+
+				for (let i = 0; i < tasks.length; i++) {
+					const executor = Promise.resolve().then(tasks[i]);
+					results.push(executor);
+
+					if (executing.length < limit) {
+						executing.push(executor);
+						executor.finally(() => {
+							const index = executing.indexOf(executor);
+							if (index > -1) {
+								executing.splice(index, 1);
+							}
+						});
+					}
+
+					if (executing.length >= limit) {
+						await Promise.race(executing);
+					}
+				}
+
+				return Promise.all(results);
+			};
+
+			// @ts-ignore
+			const chunkSize = this.syncConfig.chunkSize ?? 100;
+			const tasks = [];
+			for (let i = 0; i < accountPublicKeys.length; i += chunkSize) {
+				const chunk = accountPublicKeys.slice(i, i + chunkSize);
+				tasks.push(() =>
+					this.connection.getMultipleAccountsInfo(chunk, {
+						commitment: this.commitment,
+					})
+				);
+			}
+
+			// @ts-ignore
+			const concurrencyLimit = this.syncConfig.concurrencyLimit ?? 10;
+			const chunkedAccountInfos = await limitConcurrency(
+				tasks,
+				concurrencyLimit
+			);
+
+			const programAccountBufferMap = new Map<string, Buffer>();
+
+			for (
+				let chunkIndex = 0;
+				chunkIndex < chunkedAccountInfos.length;
+				chunkIndex++
+			) {
+				const accountInfos = chunkedAccountInfos[chunkIndex];
+				if (!accountInfos) continue;
+
+				for (let index = 0; index < accountInfos.length; index++) {
+					const accountInfo = accountInfos[index];
+					if (accountInfo === null) continue;
+
+					const publicKeyIndex = chunkIndex * chunkSize + index;
+					const publicKeyString = accountPublicKeys[publicKeyIndex].toString();
+					const buffer = Buffer.from(accountInfo.data);
+					programAccountBufferMap.set(publicKeyString, buffer);
+
+					const decodedUser = this.decode('User', buffer);
+
+					const currAccountWithSlot = this.getWithSlot(publicKeyString);
+					if (
+						currAccountWithSlot &&
+						currAccountWithSlot.slot <= accountInfo.lamports
+					) {
+						this.updateUserAccount(
+							publicKeyString,
+							decodedUser,
+							accountInfo.lamports
+						);
+					} else {
+						await this.addPubkey(
+							new PublicKey(publicKeyString),
+							decodedUser,
+							accountInfo.lamports
+						);
+					}
+				}
+			}
```

Comment:
does it make sense to put this into the task? so you also decode in parallel when individual RPC requests finish